service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/AbstractServiceClient.java

```diff
@@ -160,6 +160,18 @@ protected String encodeUrl(String url) {
     protected String getOptURI(String paramName, Object paramValue) {
         if (paramValue == null || paramValue instanceof String && paramValue.toString().isEmpty()) {
             return "";
+        } else if (paramValue instanceof List) {
+            StringBuilder builder = new StringBuilder();
+            String delimiter = "";
+            List<String> paramValueList = (List<String>) paramValue;
```

Comment:
We can't assume its always List<String> here. Param could be List<Integer>, List<Long> etc.